### DISCUSSION TITLE
Remix application not caching page or loader data reliably

### DISCUSSION THREAD

**Discussion by bradgarropy** [vercel/community#5107] [Help]
### Summary

# Background

I have a [Remix application](https://github.com/bradgarropy/bradgarropy.com/tree/main) that reaches out to the YouTube API each time the [home page](https://bradgarropy.com/) is rendered. The YouTube API imposes a daily quota, and I am trying to use caching to not make an API call to YouTube every time the page is loaded.

# What's Happening

I placed `Cache-Control` headers in both the [loader](https://github.com/bradgarropy/bradgarropy.com/blob/main/src/routes/_index.tsx#L12) and in the [headers](https://github.com/bradgarropy/bradgarropy.com/blob/main/src/routes/_index.tsx#L34) function of the home page.

```json
{"Cache-Control": "public, s-maxage=86400"}
```

I would have expected that the home page would be cached for one day (`86400 seconds`), and that the YouTube API would not have been called. However, while monitoring my YouTube API quota, the numbers keep going up.

![image](https://github.com/vercel/community/assets/11336745/7747da3f-0264-4cd8-bf1d-b6354822be76)

# Investigation

While visiting the home page from my machine, I see what I expect, an `X-Vercel-Cache: HIT` header.

I set up a [log filter](https://vercel.com/bradgarropy/bradgarropy/logs?requestPaths=%2F&timeline=maximum&vercelCaches=MISS&vercelCaches=NONE&preset=c1d6d102a39474dd4cc653e51e630a1ef403a4026c4dfdf1c1084331f0a1cf49_u&page=1&startDate=1701959862887&endDate=1701963462887) to show anytime a cache value is `MISS` or `NONE` and I do get a few every hour, which explains why the YouTube API is getting hit.

But why are we having cache misses? Isn't it a shared cache among all users?

# Resources

[Source code](https://github.com/bradgarropy/bradgarropy.com/blob/main/src/routes/_index.tsx)
[Live demo](https://bradgarropy.com/)
[Vercel Remix Docs](https://vercel.com/docs/frameworks/remix#cache-control-headers)
[Vercel Caching Docs](https://vercel.com/docs/edge-network/caching)
[Related issue](https://github.com/orgs/vercel/discussions/1559)

### Example

https://bradgarropy.com

### Steps to Reproduce

1. Visit the [home page](https://bradgarropy.com/) of my website.
2. Inspect the `X-Vercel-Cache` header.
3. Inspect the [logs](https://vercel.com/bradgarropy/bradgarropy/logs?requestPaths=%2F&timeline=maximum&vercelCaches=MISS&vercelCaches=NONE&preset=c1d6d102a39474dd4cc653e51e630a1ef403a4026c4dfdf1c1084331f0a1cf49_u&page=1&startDate=1701959862887&endDate=1701963462887) for cache misses.

**Comment by ryanflorence** [comment]
One thing to remember with cache-control is that the edge network that's caching these documents is a bunch of different data centers. If somebody from Europe visits your page and then somebody from San Francisco, it's unlikely to be sharing the same cache, so you may just be seeing cache misses for the various regions.

Cache-Control's utility increases dramatically with high traffic as all the data centers fill up, and the reverse is true for lower traffic.
